Mathlib/Algebra/DirectSum/Internal.lean

```diff
@@ -337,6 +346,90 @@ theorem DirectSum.coeAlgHom_of [AddMonoid ι] [CommSemiring S] [Semiring R] [Alg
 
 end DirectSum
 
+/-! ### Facts about grade zero -/
+
+namespace SetLike.GradeZero
+
+section Semiring
+variable [Semiring R] [AddMonoid ι] [SetLike σ R] [AddSubmonoidClass σ R]
+variable (A : ι → σ) [SetLike.GradedMonoid A]
+
+/-- The subsemiring `A 0` of `R`. -/
+def subsemiring : Subsemiring R where
+  carrier := A 0
+  __ := submonoid A
+  add_mem' := add_mem
+  zero_mem' := zero_mem (A 0)
+
+/-- The semiring `A 0` inherited from `R` in the presence of `SetLike.GradedMonoid A`. -/
+instance instSemiring : Semiring (A 0) := (subsemiring A).toSemiring
+
+/-- The linter message "error: SetLike.GradeZero.coe_natCast.{u_4, u_2, u_1} Left-hand side
+  does not simplify, when using the simp lemma on itself." is wrong. The LHS does simplify. -/
+@[nolint simpNF, simp, norm_cast] theorem coe_natCast (n : ℕ) : (n : A 0) = (n : R) := rfl
+
+/-- The linter message "error: SetLike.GradeZero.coe_ofNat.{u_4, u_2, u_1} Left-hand side does
+  not simplify, when using the simp lemma on itself." is wrong. The LHS does simplify. -/
+@[nolint simpNF, simp, norm_cast] theorem coe_ofNat (n : ℕ) [n.AtLeastTwo] :
+    (no_index (OfNat.ofNat n) : A 0) = (OfNat.ofNat n : R) := rfl
```

Comment:
So is this actually a bug in the simpNF linter which needs to be reported?

Reply:
Yes, but I think it's not the only such one.